internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -878,6 +900,9 @@ func (c *Coordinator) runLoopIteration(ctx context.Context) {
 	case overrideState := <-c.overrideStateChan:
 		c.setOverrideState(overrideState)
 
+	case upgradeDetails := <-c.upgradeDetailsChan:
+		c.setUpgradeDetails(upgradeDetails)
```

Comment:
Should we be logging every upgrade details transition? This seems like great way to make upgrade progress obvious in our logs.

Fleet is only going to get the latest state at the time of the next checkin, it won't see the entire set of transitions.

Reply:
Good idea, will add logging to this PR.

Reply:
I added logging but I implemented it as just another observer that's registered with the `UpgradeDetails` object.  See d2a2eff52.

Here's what the result looks like in Agent logs:

```
$ sudo elastic-agent logs -f | grep -i upgrade_details
{"log.level":"info","@timestamp":"2023-10-17T20:40:11.633Z","log.origin":{"file.name":"coordinator/coordinator.go","file.line":484},"message":"updated upgrade details","log":{"source":"elastic-agent"},"upgrade_details":{"target_version":"8.13.0","state":"UPG_REQUESTED","metadata":{"scheduled_at":"0001-01-01T00:00:00Z","failed_state":"","error_msg":""}},"ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2023-10-17T20:40:11.633Z","log.origin":{"file.name":"coordinator/coordinator.go","file.line":484},"message":"updated upgrade details","log":{"source":"elastic-agent"},"upgrade_details":{"target_version":"8.13.0","state":"UPG_DOWNLOADING","metadata":{"scheduled_at":"0001-01-01T00:00:00Z","failed_state":"","error_msg":""}},"ecs.version":"1.6.0"}
{"log.level":"info","@timestamp":"2023-10-17T20:40:31.594Z","log.origin":{"file.name":"coordinator/coordinator.go","file.line":484},"message":"updated upgrade details","log":{"source":"elastic-agent"},"upgrade_details":{"target_version":"8.13.0","state":"UPG_FAILED","metadata":{"scheduled_at":"0001-01-01T00:00:00Z","failed_state":"UPG_DOWNLOADING","error_msg":"failed download of agent binary: context canceled"}},"ecs.version":"1.6.0"}
```

BTW, that last entry is from me Ctrl-C'ing the `sudo elastic-agent upgrade 8.13.0` command since I knew it would never succeed.